### PR TITLE
Revised true and false operators (and literals)

### LINKED ISSUE #8444: Logical Error?
@yiwangwenrou commented on [Wed Oct 17 2018](https://github.com/dotnet/docs.zh-cn/issues/219)

DBBool TRUE = DBBool.True,NULL = DBBool.Null;    if(NULL && TRUE)  与 if(!(NULL && TRUE)   都为逻辑true

---
#### 文档详细信息

⚠ *请勿编辑此部分。 docs.microsoft.com ➟ GitHub 问题链接必须具有此部分。*

* ID: f818b111-bdeb-96a1-721a-1be183519f6d
* Version Independent ID: e98c5f3f-24d8-ae0e-d4b8-5607ce40f6b9
* Content: [false 运算符（C# 参考）](https://docs.microsoft.com/zh-cn/dotnet/csharp/language-reference/keywords/false-operator#see-also)
* Content Source: [docs/csharp/language-reference/keywords/false-operator.md](https://github.com/dotnet/docs.zh-cn/blob/live/docs/csharp/language-reference/keywords/false-operator.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---

@imba-tjd commented on [Wed Oct 17 2018](https://github.com/dotnet/docs.zh-cn/issues/219#issuecomment-430613297)

I actually got `DBBool.Null`, both of them.
default

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #8444

As I was around, I also went through the articles about `true`/`false` literals.

Supported by dotnet/samples#493 (thanks to @BillWagner for the example)

### CODE REVIEW

**Review by BillWagner — COMMENTED** [review]
This looks great @pkulikov 

I'll add the "Hold for related PR" label, and give it a final review after we've merged the associated samples.

Thanks again for contributing to docs!

**Comment by BillWagner** [comment]
closing and reopening to start a new build

**Review by BillWagner — CHANGES_REQUESTED** [review]
This looks great @pkulikov 

We need to fix the links before I :shipit:.

I had one other thought for a (possible) later PR, but I'd like your thoughts first.

File: docs/csharp/language-reference/keywords/true-operator.md [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
   - "true operator [C#]"
 ms.assetid: acaba817-5da5-4364-b3b2-2e5c75ec1839
 ---
 # true operator (C# Reference)
 
-Returns the [bool](bool.md) value `true` to indicate that an operand is true and returns `false` otherwise.
+Returns the [bool](bool.md) value `true` to indicate that an operand is definitely true. If a type defines the `true` operator, it must also define the [false operator](false-operator.md). The `true` and `false` operators are not guaranteed to complement each other.
 
-Prior to C# 2.0, the `true` and `false` operators were used to create user-defined nullable value types that were compatible with types such as `SqlBool`. However, the language now provides built-in support for nullable value types, and whenever possible you should use those instead of overloading the `true` and `false` operators. For more information, see [Nullable Types](../../programming-guide/nullable-types/index.md).
+For the example that defines both `true` and `false` operators, see the [false operator](false-operator.md) article.
```
[the remark this answers is not in the document]
Comment (pkulikov) [reply docs/csharp/language-reference/keywords/true-operator.md]: Completely agree with that! Yes, let's make it another PR. I will create one.

**Comment by pkulikov** [comment]
@BillWagner thanks for review, I've addressed your feedback

**Review by BillWagner — APPROVED** [review]

**Pull request merged**